CDRIVER-6274 finalize client backpressure phase 1 rollout (#2269)
mongodb/mongo-c-driver: src/libmongoc/doc/mongoc_uri_t.rst

MONGOC_URI_RETRYREADS                      retryreads                        true                              If "true" and the server is a MongoDB 3.6+ standalone, replica set, or sharded cluster, the driver safely retries a read that failed due to a network error or replica set failover.
MONGOC_URI_RETRYWRITES                     retrywrites                       true if driver built w/ TLS       If "true" and the server is a MongoDB 3.6+ replica set or sharded cluster, the driver safely retries a write that failed due to a network error or replica set failover. Only inserts, updates of single documents, or deletes of single
                                                                                                               documents are retried.
MONGOC_URI_MAXADAPTIVERETRIES              maxadaptiveretries                2                                 The maximum number of times the driver retries an operation that fails with a `SystemOverloadedError`. Must be a non-negative integer.

> MONGOC_URI_ENABLEOVERLOADRETARGETING       enableoverloadretargeting         false                             If "true", the driver deprioritizes a server that returns a `SystemOverloadedError` when selecting a server for retry.

Reviewer: ```suggestion
MONGOC_URI_ENABLEOVERLOADRETARGETING       enableoverloadretargeting         false                             If "true", the driver deprioritizes servers which returned a `SystemOverloadedError` when selecting a server to retry an operation.
```
Minor wording tweak suggestion.
